### ISSUE TITLE
`required_version` is too strict with `0.39.0` 

### ISSUE THREAD

**Issue by franzs** [terraform-linters/tflint#1453]
## Introduction

The new version `0.39.0` of `tflint` causes extra warnings.

## Expected Behavior

No warning should occur as with version `0.38.1` and below.

## Actual behavior

Having a `terraform` block without `required_version` causes a warning although there is a `required_version` in another `terraform` block:

```console
% tflint
1 issue(s) found:

Warning: terraform "required_version" attribute is required (terraform_required_version)

  on  line 0:
   (source code not available)

Reference: https://github.com/terraform-linters/tflint/blob/v0.39.0/docs/rules/terraform_required_version.md
```

## Step to Reproduce

`backend.tf`:

```hcl
terraform {
  backend "gcs" {
    bucket = "tfstate"
    prefix = "prefix"
  }
}
```

`versions.tf`:

```hcl
terraform {
  required_version = "~> 1.1.0"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 4.27.0"
    }
  }
}
```
Run `tflint` version `0.39.0`.

## Additional Context

```console
% tflint -v
TFLint version 0.39.0
% terraform -v
Terraform v1.1.9
on linux_amd64
```

**Comment by franzs** [comment]
Oh, this is a duplcate of #1451. Sorry...
